src/message-queues.ts

```diff
@@ -142,9 +146,15 @@ export abstract class MessageQueue {
       this._subscriber.emit('error', e);
     }
 
+    this.numInFlightRequests -= batchSize;
     if (deferred) {
       deferred.resolve();
     }
+
+    if (this.numInFlightRequests === 0 && this._onDrain) {
```

Comment:
Oh, super minor thing here that I thought @bcoe was going to comment on - to be overly defensive, it might be nice to make this `<= 0`.

Reply:
Done! :)